src/v/cloud_storage/recursive_directory_walker.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "seastarx.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/core/gate.hh>
+#include <seastar/core/lowres_clock.hh>
+
+#include <vector>
+
+namespace cloud_storage {
+struct file_list_item {
+    ss::lowres_clock::time_point access_time;
+    ss::sstring filename;
+    uint64_t size;
+};
+class recursive_directory_walker {
+public:
+    ss::future<> stop();
+    ss::future<> walk(ss::sstring start_dir);
+    uint64_t get_current_cache_size();
+    std::vector<file_list_item> get_files();
```

Comment:
These should be const. Or even better, because these fields get nuked every time we run walk, I think they should just be returned from walk.

Reply:
Just deleted this functions :)

Reply:
Now we can delete the fields too :)